### ISSUE TITLE
Typora does not find files in PATH needed for custom Pandoc exports

### ISSUE THREAD

**Issue by tweedyf** [typora/typora-issues#5025]
- [x] Searched existing issues to avoid creating duplicates.

- [x] Confirmed that it can be reproduced in built-in themes without customized css.

   If it only exists in 3rd party themes or css, you can still report it, but please attach the theme target or the css file. We may not "fix" it, if it is caused by 3rd party themes or css styles that we cannot support.

- [x] Searched http://support.typora.io/

----

**Describe the bug**
When I try to export to PDF via Pandoc, either via a LaTeX pdf generator or via groff/pdfroff, the export fails because it cannot find related programs (`extractbb` for LaTeX; `ghostscript` for groff). This happens regardless of whether /opt/homebrew/bin is in my shell path or in the systemwide path at /etc/paths.

System notes: 
I have Pandoc, LaTeX, and Groff installed on an M1 Mac via Homebrew; so, they're all in /opt/homebrew/bin. Built-in Pandoc exports work fine. Typora finds Pandoc in the homebrew bin fine. Setting --pdf-engine=full/path/to/pdfgenerator doesn't work, because it fails to find supporting programs required by those pdf generators.

Here are the errors when I specify the full path to the pdf engine:
```
pdfroff: installation problem: cannot find GhostScript interpreter

*** FATAL INSTALLATION ERROR ***

'pdfroff' requires a GhostScript interpreter to convert PostScript to PDF.
Since you do not appear…

args: [-f, native, -s, -o, /Users/tweedy/Downloads/Futile Resistance 12.pdf, -t, ms, --pdf-engine=/opt/homebrew/bin/pdfroff]
```

```
sh: extractbb: command not found
sh: extractbb: command not found
Error producing PDF.
! File ended while scanning use of \__sys_tmp:w.
<inserted text> 
\par 
l.148   { \sys_load_backe…

args: [-f, native, -s, -o, /Users/tweedy/Downloads/Futile Resistance 12.pdf, --pdf-engine=/opt/homebrew/bin/xelatex]
```

**To Reproduce**
1. Add custom Pandoc export
2. Set "Add from template" to Pandoc (issue also happens if you select PDF (Latex/Pancoc) [sic])
3. Use pdflatex, xelatex, or pdfroff as pdf engine
4. Attempt to export

**Expected behavior**
Typora should find the dependencies in PATH and allow them to be called when performing custom Pandoc exports.

**Sample Markdown File**
Not dependent on any particular markdown file.

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.1

**Typora Version**
1.0.5 (6032)

**Comment by abnerlee** [comment]
Typora uses PATH of current process. homebrew usually change PATH in .bashrc which only affects your shell. (https://superuser.com/questions/324616/how-should-i-set-the-path-variable-on-my-mac-so-the-hombrew-installed-tools-are)

But Typora will add brew's path into its PATH env as well from next version up.

**Comment by tweedyf** [comment]
> This happens regardless of whether /opt/homebrew/bin is in my shell path **or in the systemwide path at /etc/paths**

Just to reiterate the point about systemwide path too, since it makes sense that Typora doesn't import ~/.bashrc (or whatever shell rc), but it doesn't make sense that it doesn't import /etc/paths

**Comment by abnerlee** [comment]
should be fixed in 1.1
